### PR TITLE
feat: Codeowners generation

### PR DESCRIPTION
**Pull request by jpmcb**
## Description

This implements an experimental, proof of concept command:

```
$ pizza generate codeowners path/to/local/repository [flags]
```

This expects a path to a valid git repository where it will iterate the commits (by default, over the last 90 days) and generate a `OWNERS` file (or a github style `CODEOWNERS` file if the `--github-codeowners` flag is provided).

This file relys on a configuration file in the user's home directory called `~/.sauced.yaml` which carries attributions of user keys to emails (generally, these are GitHub users or teams but could be simple names). For example:

Individual user attribution:

```yaml
attribution:
  jpmcb:
    - john@opensauced.pizza
```

Team attribution:

```yaml
attribution:
  open-sauced/engineering:
    - john@opensauced.pizza
```

Agnostic name attribution:

```yaml
attribution:
  John McBride:
    - john@opensauced.pizza
```

## Related Tickets & Documents

Closes: https://github.com/open-sauced/pizza-cli/issues/85

## Steps to QA

1. Build via `just build`
2. Run the codeowners command on some local git repo via `./build/pizza generate codeowners .` (make sure you have a valid yaml config in `~/.sauced.yaml`

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

### CODE REVIEW

**Comment by brandonroberts** [comment]
I'm partial, but I think the GitHub style should be the default, as there may be more GitHub style repos than Google/Kubernetes style repos in terms of owners.

**Comment by jpmcb** [comment]
> I think the GitHub style should be the default

Sure thing: easy enough to flip that bool flag around 👍🏼 I think this is what I was doing originally, but this was feedback we got from our stakeholder (but their probably partial to the Google style `OWNERS` file anyways). Let's go with the GitHub style as default since that's more what we'll be targeting for this as it goes out.

File: README.md [hunk 1]
```diff
@@ -2,70 +2,69 @@
...
+    codeowners /app/workspace/gopherlogs -c /app/.sauced.yaml
+```
+
+### Go install
+
+Using the Go tool-chain, you can install the binary directly:
 
 ```sh
-brew install open-sauced/tap/pizza
+$ go install github.com/open-sauced/pizza-cli
```
Comment (nickytonline) [inline README.md:51]: I ran this and it failed. It appears to required a version number unless there's something else about the Go ecosystem I'm unaware of.

```
❯ go install github.com/open-sauced/pizza-cli       
go: 'go install' requires a version when current directory is not in a module
        Try 'go install github.com/open-sauced/pizza-cli@latest' to install the latest version

❯ go install github.com/open-sauced/pizza-cli@v1.1.1
go: downloading github.com/open-sauced/pizza-cli v1.1.1

~/dev/work took 3s 
```

Comment (jpmcb) [reply README.md:51]: Whoops! Forgot `@lateset` for the module version.

**Comment by jpmcb** [comment]
This now generates a `CODEOWNERS` by default:

```
❯ ./build/pizza generate codeowners .
Finished processing commits for: /Users/jpmcb/workspace/opensauced/pizza-cli
Finished generating file: /Users/jpmcb/workspace/opensauced/pizza-cli/CODEOWNERS
```

and the documentation reflects using the `latest` for the Go module install method and the docker run method.

Ready to roll!

**Review by nickytonline — APPROVED** [review]
:shipit:

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**